internal/kafka/internal/workers/cluster_mgrs/cleanup_clusters_mgr.go

```diff
@@ -0,0 +1,113 @@
+package cluster_mgrs
+
+import (
+	fleeterrors "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/services/sso"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/services"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/workers"
+	"github.com/google/uuid"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/metrics"
+	"github.com/golang/glog"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	cleanupClustersWorkerType = "cleanup_clusters"
+)
+
+// CleanupClustersManager represents a worker that periodically reconciles data plane clusters in cleanup state.
+type CleanupClustersManager struct {
+	workers.BaseWorker
+	clusterService             services.ClusterService
+	kasFleetshardOperatorAddon services.KasFleetshardOperatorAddon
+	osdIdpKeycloakService      sso.OsdKeycloakService
+}
+
+// NewCleanupClustersManager creates a new worker that reconciles data plane clusters in cleanup state.
+func NewCleanupClustersManager(reconciler workers.Reconciler,
+	clusterService services.ClusterService,
+	kasFleetshardOperatorAddon services.KasFleetshardOperatorAddon,
+	osdIdpKeycloakService sso.OsdKeycloakService) *CleanupClustersManager {
+	return &CleanupClustersManager{
+		BaseWorker: workers.BaseWorker{
+			Id:         uuid.New().String(),
+			WorkerType: cleanupClustersWorkerType,
+			Reconciler: reconciler,
+		},
+		clusterService:             clusterService,
+		kasFleetshardOperatorAddon: kasFleetshardOperatorAddon,
+		osdIdpKeycloakService:      osdIdpKeycloakService,
+	}
+}
+
+// Start initializes the worker to reconcile data plane clusters in cleanup state.
+func (m *CleanupClustersManager) Start() {
+	m.StartWorker(m)
+}
+
+// Stop causes the process for reconciling data plane clusters in cleanup state to stop.
+func (m *CleanupClustersManager) Stop() {
+	m.StopWorker(m)
+}
+
+func (m *CleanupClustersManager) Reconcile() []error {
+	glog.Infoln("reconciling clusters")
+
+	var errList fleeterrors.ErrorList
+	err := m.processCleanupClusters()
+	if err != nil {
+		errList.AddErrors(err)
+	}
+
+	return errList.ToErrorSlice()
```

Comment:
Why do we convert it to a slice? I believe the function should return the `error` interface, and it will be up to the caller to cast it to `ErrorList` if he needs the embedded error list.

Reply:
The reason we convert it to an empty slice is to avoid refactoring the caller code in this PR, as `Reconcile` is an interface method and changing the signature would mean refactoring all the workers in this PR

Reply:
The `.Reconciler` signatures needs a slice hence the conversion to a slice.
Changing the signature of that method requires a whole lot of other modification which are not part of this PR.